lib/availability_resolver.js

```diff
@@ -102,76 +103,71 @@ class AvailabilityResolver {
       })
   }
 
-  _fixItemRequestabilityInResponse (barcodesAndAvailability, request) {
+  _fixItemRequestability (request) {
     for (let hit of this.elasticSearchResponse.hits.hits) {
       (hit._source.items || []).map((item) => {
-        // If it's an electronic item, skip it
         if (item.electronicLocator) return item
-        const barcode = this._getItemBarcode(item)
-
-        // It's in ReCAP if
-        //  1. it has a ReCap customer code
-        //  2. it's an NYPL item with a ReCAP holding location or
-        //  3. it's not an NYPL owned item (i.e. it's a partner item)
-        const isInRecap = this._isInRecap(item)
-        item.physRequestable = false
         item.eddRequestable = false
+        item.physRequestable = false
         item.specRequestable = false
+        item.requestable = [false]
+        const isInRecap = this._isInRecap(item)
+        item.eddRequestable = this._fixEddRequestability(item, isInRecap, request)
+        item.physRequestable = this._fixPhysRequestability(item, isInRecap)
+        item.specRequestable = !!item.aeonUrl
+        if (item.eddRequestable || item.physRequestable || item.specRequestable) {
+          item.requestable[0] = true
+        } else item.requestable[0] = false
+        return item
+      })
+    }
+  }
+  _fixEddRequestability (item, isInRecap, request) {
+    // If it's a recap item, determine edd based on code
+    if (isInRecap) {
+      if (Array.isArray(item.recapCustomerCode) && recapCustomerCodes[item.recapCustomerCode[0]]) {
+        return recapCustomerCodes[item.recapCustomerCode[0]].eddRequestable
+      } else {
+        return false
+      }
+    }
+    if (Feature.enabled('no-on-site-edd', request)) {
+      return false
+    } else {
+      // If it's onsite, use different criteria
+      return DeliveryLocationsResolver.eddRequestableByOnSiteCriteria(item)
+    }
+  }
 
-        // If it's in ReCAP
+  _fixPhysRequestability (item, isInRecap) {
+    // this is not the final physRequestable value. True physRequestable is
+    // only determined on the front end via the vars closed_locations and
+    // open_locations.
+    if (isInRecap) {
+      return DeliveryLocationsResolver.__deliveryLocationsByCustomerCode.length > 0
```

Comment:
This is a function but you're calling `.length` on it without calling it?

Reply:
updated to invoke the function